File: server/src/main/kotlin/pt/isel/g20/unicommunity/forum/model/Forum.java
package pt.isel.g20.unicommunity.forum.model;

import javax.persistence.*;

@Entity
public class Forum {

    @Id
    private long boardId;

    private boolean allowImagePosts;

    public Forum(){}

    public Forum(Long boardId){
        this(boardId, false);
    }

    public Forum(Long boardId, boolean allowImagePosts){
        this.boardId = boardId;
        this.allowImagePosts = allowImagePosts;
    }

    public long getBoardId() {
        return boardId;
    }

    public void setBoardId(long boardId) {
        this.boardId = boardId;
    }

    public boolean isAllowImagePosts() {
        return allowImagePosts;
    }

    public void setAllowImagePosts(boolean allowImagePosts) {
        this.allowImagePosts = allowImagePosts;
    }
}

File: server/src/main/kotlin/pt/isel/g20/unicommunity/forum/model/Forum.kt
package pt.isel.g20.unicommunity.forum.model

import pt.isel.g20.unicommunity.board.model.Board
import pt.isel.g20.unicommunity.forumItem.model.ForumItem
import javax.persistence.*

@Entity
class Forum(
        @Id
        var id: Long,

        var isAllowImagePosts: Boolean
) {

    @OneToOne
    var board: Board? = null

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn
    var items: MutableList<ForumItem> = mutableListOf()

    constructor() : this (0, false)

    constructor(id: Long): this(id, false)
}
